src/lib/db/feature-toggle-admin-store.ts

```diff
@@ -0,0 +1,287 @@
+import { Knex } from 'knex';
+import metricsHelper from '../util/metrics-helper';
+import { DB_TIME } from '../metric-events';
+import { Logger, LogProvider } from '../logger';
+import {
+    FeatureToggle,
+    IFeatureToggleClient,
+    IFeatureToggleQuery,
+    IStrategyConfig,
+    ITag,
+} from '../types/model';
+import { DEFAULT_ENV } from '../util/constants';
+import { PartialDeep } from '../types/partial';
+import EventEmitter from 'events';
+import FeatureToggleStore from './feature-toggle-store';
+import { ensureStringValue } from '../util/ensureStringValue';
+import { mapValues } from '../util/map-values';
+import { Db } from './db';
+import { IFeatureToggleAdminStore } from '../types/stores/feature-toggle-admin-store';
+import Raw = Knex.Raw;
+
+export interface IGetAllFeatures {
+    featureQuery?: IFeatureToggleQuery;
+    archived: boolean;
+    includeStrategyIds?: boolean;
+    userId?: number;
+}
+
+export interface IGetAdminFeatures {
+    featureQuery?: IFeatureToggleQuery;
+    archived?: boolean;
+    userId?: number;
+}
+
+export default class FeatureToggleAdminStore
+    implements IFeatureToggleAdminStore
+{
+    private db: Db;
+
+    private logger: Logger;
+
+    private timer: Function;
+
+    constructor(db: Db, eventBus: EventEmitter, getLogger: LogProvider) {
+        this.db = db;
+        this.logger = getLogger('feature-toggle-admin-store.ts');
+        this.timer = (action) =>
+            metricsHelper.wrapTimer(eventBus, DB_TIME, {
+                store: 'feature-toggle',
+                action,
+            });
+    }
+
+    private async getAll({
+        featureQuery,
+        archived,
+        userId,
+    }: IGetAllFeatures): Promise<(IFeatureToggleClient & FeatureToggle)[]> {
```

Comment:
we could remove 50% of the code and not return IFeatureToggleClient specific stuff if the segments.e2e.test.ts wasn't using admin endpoint to fetch strategies that are only part of the client API. Are there any reasons why segments tests do that? @chriswk @sighphyre